File: application/core/src/main/java/br/com/beblue/bluerecords/core/usecase/VendaUseCaseImpl.java
package br.com.beblue.bluerecords.core.usecase;

import br.com.beblue.bluerecords.core.command.ConsultaVendaCommand;
import br.com.beblue.bluerecords.core.command.RegistraVendaCommand;
import br.com.beblue.bluerecords.core.entitidade.Venda;
import br.com.beblue.bluerecords.core.entitidade.paginacao.Paginacao;
import br.com.beblue.bluerecords.core.repositorio.VendaRepositorio;

public class VendaUseCaseImpl implements VendaUseCase {

    private VendaRepositorio vendaRepositorio;
    private NotificadorVenda notificadorVenda;

    public VendaUseCaseImpl(VendaRepositorio vendaRepositorio, NotificadorVenda notificadorVenda) {
        this.vendaRepositorio = vendaRepositorio;
        this.notificadorVenda = notificadorVenda;
    }

    @Override
    public Venda registrarVenda(RegistraVendaCommand command) {
        Venda venda = vendaRepositorio.cadastrar(command);
        if(venda.getId()>0)
            notificadorVenda.notificarNovaVenda(venda.getId());

        return venda;
    }

    @Override
    public Paginacao<Venda> consultarVendas(ConsultaVendaCommand command) {
        return vendaRepositorio.consultaVendas(command);
    }

    @Override
    public Venda consultarVendaPorId(Integer id) {
        return vendaRepositorio.consulta(id);
    }

}

File: application/core/src/main/kotlin/br/com/beblue/bluerecords/core/usecase/venda/VendaUseCaseImpl.kt
package br.com.beblue.bluerecords.core.usecase.venda

import br.com.beblue.bluerecords.core.entity.Venda
import br.com.beblue.bluerecords.core.repository.VendaRepositorio
import br.com.beblue.bluerecords.core.usecase.venda.command.ConsultaVendasCommand
import br.com.beblue.bluerecords.core.usecase.venda.command.RegistrarVendaCommand

class VendaUseCaseImpl(
    private val vendaRepositorio: VendaRepositorio
) : VendaUseCase {

    override fun registrarVenda(command: RegistrarVendaCommand): Venda {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun consultarVendas(command: ConsultaVendasCommand): List<Venda> {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun consultarVendaPorId(id: Int): Venda {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

}